### PR TITLE
ui(InstallInstructions): add error handling

### PR DESCRIPTION
**Pull request by MoSattler**
Error handling of the "Install Dashboard" button

Screenshot 2021-07-29 at 17 27 17

# Have you...

* [x] Added an explanation of what your changes do?
* [x] Written new tests for your changes?
* [x] Thought about which docs need updating?

### CODE REVIEW

File: packages/app/src/client/integrations/common/UninstallIntegrationBtn.test.tsx [hunk 1]
```diff
@@ -163,7 +163,6 @@ test("shows error messages if grafana request fails", async () => {
     rest.delete(
       `http://${tenant.name}.localhost/grafana/api/folders/i9n-${integration.id}`,
       (req, res, ctx) => {
-        ctx.status(500);
```
Comment (MoSattler) [inline packages/app/src/client/integrations/common/UninstallIntegrationBtn.test.tsx:166]: Cleanup - unused code that doesn't do anything

File: packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx [hunk 2]
```diff
@@ -71,6 +72,33 @@ export const InstallInstructions = ({
...
+        state: "error" as const,
+        title,
+        information: message,
+        handleClose: () =>
+          unregisterNotification({
+            id: messageId,
+            title: "",
+            information: ""
+          })
+      };
```
Comment (MoSattler) [inline packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx:96]: I would like to at some point spend some time simplifying the notification service interface. There is a bunch of stuff in the interface that is not really used, but yet required.

File: packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx [hunk 3]
```diff
@@ -89,13 +117,45 @@ export const InstallInstructions = ({
...
   const dashboardHandler = async () => {
-    const folder = await grafana.createFolder({ integration, tenant });
+    let folder;
+    try {
+      folder = await grafana.createFolder({ integration, tenant });
+    } catch (error) {
+      notifyError(
+        `Could not create grafana integration dashboard folder ${integration.name}`,
+        error.response.data.message ?? error.message
+      );
```
Comment (MoSattler) [inline packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx:127]: Real world example:
Screenshot 2021-07-29 at 16 45 53

File: packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx [hunk 4]
```diff
@@ -89,13 +117,45 @@ export const InstallInstructions = ({
...
+         */
+        const errorMessageList = Array.isArray(error.response.data)
+          ? error.response.data.map((errObj: Object) => JSON.stringify(errObj))
+          : [error.message];
+
+        for (const message of errorMessageList) {
+          notifyError(
+            `Could not create grafana integration dashboard ${d.uid}`,
+            message
+          );
```
Comment (MoSattler) [inline packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx:155]: Real world example
Screenshot 2021-07-29 at 17 10 30

File: packages/app/src/client/integrations/k8sMetrics/Show/dashboards/index.ts [hunk 5]
```diff
@@ -27,7 +27,7 @@ type DashboardProps = {
 export function makePrometheusDashboardRequests({
   integrationId,
   folderId
-}: DashboardProps): object[] {
```
Comment (MoSattler) [inline packages/app/src/client/integrations/k8sMetrics/Show/dashboards/index.ts:30]: Small clean up. It's better to have no typing than bad generic typing, as non-explicit typing can often be inferred by TS.

File: packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx (hunk 4 above)
Comment (MoSattler) [reply packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx:155]: Not the best error message, but might be understandable.

File: packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx (hunk 2 above)
Comment (MatApple) [reply packages/app/src/client/integrations/k8sMetrics/Show/InstallInstructions.tsx:96]: 👍

**Pull request merged**